octoprint_mrbeam/gcodegenerator/job_params.py

```diff
@@ -0,0 +1,30 @@
+class JobParams:
```

Comment:
I would probably register this as a setting with `octoprint.settings.settings("mrbeam.job")` or so. It should then be possible to call all of it from the front end with `self.settingsViewModel.settings.plugins.mrbeam.job` or something like that.

I'm not quite sure it works, but in the worst case you can add this to the `self.on_default_settings` from `octoprint_mrbeam/__init__.py`, and that should work.

All of the leaves of the settings datastruct should be `ko.observables` so you don't have to declare everything 3 extra times in the code.

See https://github.com/mrbeam/MrBeamPlugin/pull/1031/commits/d5caf5bd91c96c0e4fbb2d632f39b2ff02de2f71#diff-f9f46dfe7b608ec38c4abe52ca160b75 for how I did it with `cam.remember_markers_across_sessions`